### ISSUE TITLE
KeyError "DB_TYPE"

### ISSUE THREAD

**Issue by akosgulyban** [cutright/DVH-Analytics#84]
Switching from version 0.6.1 to 0.8.1, I received in the console the following error:
" 
File "C:\DVHanalytics\DVH-Analytics-master0.8.1\dvha\options.py", line 329, in db_group_upgrade
    new_value = loaded_options[key]

KeyError: 'DB_TYPE'
"

**Comment by akosgulyban** [comment]
I just tried with the Windows executable as well, it provided the same error.
![DVH_analytics_error_20200604](https://user-images.githubusercontent.com/8821435/83746171-025d6300-a65f-11ea-8050-0f9bdefdd460.png)

**Comment by cutright** [comment]
If you don't mind losing your options, you can rename or delete `~/Apps/dvh_analytics/preferences/.options` and default values will be used.

This would clear your DB connection settings and visual preferences.  This will not affect anything in your database or ROI maps.  If you don't want to lose your options file, just rename the file and when I release DVHA `v0.8.2` this will be resolved when you rename it back.

I would like to fix this bug though. I'll take a closer look today.

**Comment by cutright** [comment]
Looks like you are using source code? Are you comfortable switching branches to `v0.8.2`?

I now check that you have `DB_TYPE` before trying to retrieve it.
https://github.com/cutright/DVH-Analytics/blob/45a25068b8bb2195b8b1a358ed0cd49d8cee726e/dvha/options.py#L323-L335

**Comment by cutright** [comment]
I believe this issue will only happen if you upgrade from DVHA `<0.6.7` directly to `0.8.1`.  I'm pretty sure if you run a DVHA version in between and do anything to cause `.options` save,  `DB_TYPE` will be created and stored in `.options`.

**Comment by akosgulyban** [comment]
I deleted the options, now the application is launched still it does not see the DB.
or it has no values (still the 0.6.1 version works)

Update (after some manual search):
1) deleting the option
2) updating the Database Connection solved the issue.

Bonus, now the executable is working as well.

**Comment by cutright** [comment]
Sorry about that, I'm guessing it would have worked without deleting the `.options` file (if you're on the 0.8.2 branch) but the default is SQLite as of this post. I think you would have just needed to change it to PostgreSQL and it would have retained your credentials.

I will set the default to PostgreSQL if you're upgrading from `<0.6.7` since that's when SQLite got support.

I'm glad it's working now.

**Comment by cutright** [comment]
Ran into some other issues with `0.6.3` when I was storing previous SQL credentials in separate files. So I broadened the try/except clause when loading saved options to prevent a crash.

https://github.com/cutright/DVH-Analytics/blob/f4122b74060041afa9ca9bfb3296709b424cce09/dvha/options.py#L220-L230

I also set the default `DB_TYPE` to pgsql if upgrading from `<0.6.7`.

https://github.com/cutright/DVH-Analytics/blob/f4122b74060041afa9ca9bfb3296709b424cce09/dvha/options.py#L324-L337

Hope to push to master branch with new executables today.
